### PR TITLE
Small gui code cleanup

### PR DESCRIPTION
**Pull request by razzeee**
## Description

Just some cleanup with the help of resharper

## Motivation and Context

Just streamlining stuff.

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by phate89** [comment]
jenkins build this please

File: xbmc/guilib/GUIWindow.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ bool CGUIWindow::icompare::operator()(const std::string &s1, const std::string &
 
 CGUIWindow::CGUIWindow(int id, const std::string &xmlFile)
 {
-  SetID(id);
+  CGUIWindow::SetID(id);
```
Comment (Montellese) [inline xbmc/guilib/GUIWindow.cpp]: Why is the class prefix needed? The call is already happening inside `CGUIWindow`.

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 2]
```diff
@@ -1613,11 +1607,6 @@ int CGUIWindowVideoBase::GetScraperForItem(CFileItem *item, ADDON::ScraperPtr &i
   return foundDirectly ? 1 : 0;
 }
 
-void CGUIWindowVideoBase::OnScan(const std::string& strPath, bool scanAll)
```
Comment (Montellese) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: Why remove this? Now there are two places in the code that both need to call `g_application.StartVideoScan()` manually.

File: xbmc/view/GUIViewControl.h [hunk 3]
```diff
@@ -20,7 +20,6 @@
  *
  */
 
-#include <string>
```
Comment (Montellese) [inline xbmc/view/GUIViewControl.h]: `std::string` is being used in this header file.

File: xbmc/windows/GUIMediaWindow.cpp [hunk 4]
```diff
@@ -100,7 +96,7 @@ CGUIMediaWindow::CGUIMediaWindow(int id, const char *xmlFile)
   m_iLastControl = -1;
   m_canFilterAdvanced = false;
 
-  m_guiState.reset(CGUIViewState::GetViewState(GetID(), *m_vecItems));
+  m_guiState.reset(CGUIViewState::GetViewState(CGUIControl::GetID(), *m_vecItems));
```
Comment (Montellese) [inline xbmc/windows/GUIMediaWindow.cpp]: `CGUIControl::GetID()` is virtual and could be overridden by any of the classes in the inheritance hierarchy of `CGUIMediaWindow`. Why should we explicitly bypass that?

**Review by Montellese — DISMISSED** [review]

File: xbmc/video/windows/GUIWindowVideoBase.cpp (hunk 2 above)
Comment (Razzeee) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: Well I don't think that it's healthy to have one line methods. Didn't feels like it adds anything of value.

File: xbmc/guilib/GUIWindow.cpp (hunk 1 above)
Comment (Razzeee) [inline xbmc/guilib/GUIWindow.cpp]: What @Paxxi said is correct.

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
jenkins build this please

File: xbmc/guilib/GUIWindow.cpp [hunk 5]
```diff
@@ -158,7 +158,7 @@ bool CGUIWindow::Load(TiXmlElement* pRootElement)
   if (!pRootElement)
     return false;
   
-  if (strcmpi(pRootElement->Value(), "window"))
+  if (_strcmpi(pRootElement->Value(), "window"))
```
Comment (garbear) [inline xbmc/guilib/GUIWindow.cpp]: Why the `_strcmpi`? I don't see this used anywhere else in the codebase

Comment (Razzeee) [inline xbmc/guilib/GUIWindow.cpp]: > ISO C reserves certain identifiers for future expansion (see here), including anything that starts with "str".
http://stackoverflow.com/a/1964847/2859813

Comment (garbear) [inline xbmc/guilib/GUIWindow.cpp]: strcmpi() is non-standard, so let's change this to !StringUtils::EqualsNoCase()

**Comment by Razzeee** [comment]
Updated @garbear

File: xbmc/video/windows/GUIWindowVideoBase.cpp (hunk 2 above)
Comment (phil65) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: I agree with montellese here, no need to remove it. better to keep those globals calls in an separate method. Easier to make subsequent changes then.

Comment (Razzeee) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: Changed

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
Can you guys look at this again?

File: xbmc/dialogs/GUIDialogContextMenu.cpp [hunk 6]
```diff
@@ -533,7 +531,7 @@ bool CGUIDialogContextMenu::OnContextButton(const std::string &type, const CFile
 CMediaSource *CGUIDialogContextMenu::GetShare(const std::string &type, const CFileItem *item)
 {
   VECSOURCES *shares = CMediaSourceSettings::GetInstance().GetSources(type);
-  if (!shares || !item) return NULL;
+  if (!shares || !item) return nullptr;
```
Comment (xhaggi) [inline xbmc/dialogs/GUIDialogContextMenu.cpp]: please break return to new line.

File: xbmc/guilib/GUIWindow.cpp [hunk 7]
```diff
@@ -83,7 +83,7 @@ bool CGUIWindow::Load(const std::string& strFileName, bool bContainsPath)
   CPerformanceSample aSample("WindowLoad-" + strFileName, true);
 #endif
 
-  if (m_windowLoaded || g_SkinInfo == NULL)
+  if (m_windowLoaded || g_SkinInfo == nullptr)
```
Comment (xhaggi) [inline xbmc/guilib/GUIWindow.cpp]: `!g_SkinInfo`

File: xbmc/guilib/GUIWindow.cpp [hunk 8]
```diff
@@ -253,9 +253,9 @@ bool CGUIWindow::Load(TiXmlElement* pRootElement)
       TiXmlElement *pControl = pChild->FirstChildElement();
       while (pControl)
       {
-        if (strcmpi(pControl->Value(), "control") == 0)
+        if (StringUtils::EqualsNoCase(pControl->Value(), "control"))
         {
-          LoadControl(pControl, NULL, CRect(0, 0, (float)m_coordsRes.iWidth, (float)m_coordsRes.iHeight));
+          LoadControl(pControl, nullptr, CRect(0, 0, (float)m_coordsRes.iWidth, (float)m_coordsRes.iHeight));
```
Comment (xhaggi) [inline xbmc/guilib/GUIWindow.cpp]: c++ style cast if you are on it?

File: xbmc/video/windows/GUIWindowVideoBase.cpp [hunk 9]
```diff
@@ -376,7 +375,7 @@ bool CGUIWindowVideoBase::ShowIMDB(CFileItemPtr item, const ScraperPtr &info2, b
     if (!needsRefresh)
       return pDlgInfo->HasUpdatedThumb();
     // check if the item in the video info dialog has changed and if so, get the new item
-    else if (pDlgInfo->GetCurrentListItem() != NULL)
+    else if (pDlgInfo->GetCurrentListItem() != nullptr)
```
Comment (xhaggi) [inline xbmc/video/windows/GUIWindowVideoBase.cpp]: please remove `!= nullptr`

File: xbmc/windows/GUIMediaWindow.cpp [hunk 10]
```diff
@@ -1811,7 +1807,7 @@ void CGUIMediaWindow::OnFilterItems(const std::string &filter)
     // to be able to select the same item as before we need to adjust
     // the path of the item i.e. add or remove the "filter=" URL option
     // but that's only necessary for folder items
-    if (currentItem.get() != NULL && currentItem->m_bIsFolder)
+    if (currentItem.get() != nullptr && currentItem->m_bIsFolder)
```
Comment (xhaggi) [inline xbmc/windows/GUIMediaWindow.cpp]: please remove `!= nullptr`

File: xbmc/windows/GUIMediaWindow.cpp [hunk 11]
```diff
@@ -1971,7 +1967,7 @@ bool CGUIMediaWindow::Filter(bool advanced /* = true */)
   if (!m_canFilterAdvanced || !advanced)
   {
     const CGUIControl *btnFilter = GetControl(CONTROL_BTN_FILTER);
-    if (btnFilter != NULL && btnFilter->GetControlType() == CGUIControl::GUICONTROL_EDIT)
+    if (btnFilter != nullptr && btnFilter->GetControlType() == CGUIControl::GUICONTROL_EDIT)
```
Comment (xhaggi) [inline xbmc/windows/GUIMediaWindow.cpp]: same here

**Review by xhaggi — DISMISSED** [review]
Please rebase your PR because most of the stuff you did in GUIWindowManager is already in.

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
jenkins build this please

**Comment by Razzeee** [comment]
well windows works but all else failed due to git not finding the tree? let's try again.

jenkins build this please

**Comment by Razzeee** [comment]
Everything but LINUX-64 build fine. 
> /home/jenkins/workspace/LINUX-64/tools/buildsteps/linux64/run-tests: line 10: 31414 Segmentation fault      (core dumped) build/kodi-test --gtest_output=xml:gtestresults.xml
Build step 'Execute shell' marked build as failure

**Comment by Razzeee** [comment]
jenkins build this please

**Pull request merged**